### ISSUE TITLE
Allow Self-Conflicting Entries in execution

### ISSUE THREAD

**Issue by Huzaifa696** [anza-xyz/agave#1025]
#### Description

This issue is to keep track of the progress of [SIMD#83](https://github.com/solana-foundation/solana-improvement-documents/pull/83). The changes as discussed on [discord](https://discord.com/channels/428295358100013066/885976714646290473/1229806685342666783) include:

1. Updating transaction execution to handle conflicting transactions within an entry.
2. Add feature-gate to allow conflicting txs in batch on replay-side - Adjustments to scheduler
3. Support feature-gate in block-production method to generate entries with self-conflicting transactions (not strictly required for SIMD feature to be activated) 

@apfitzge , should this be split over separate PRs? We can start with point 1 and follow up with you for review. In the meanwhile, can you create the feature-gate for 2?

References:
Issue: https://github.com/solana-labs/solana/issues/33899
Old PR: https://github.com/solana-labs/solana/pull/34104

**Comment by apfitzge** [comment]
@Huzaifa696, yes this should ideally be split across a few PRs. The above 3 points seem reasonable ways to split up the PRs.

I updated this issue title, since this will not enable bankless leader yet.
I will create a feature issue for the SIMD using this pubkey: `ENTRYnPAoT5Swwx73YDGzMp3XnNH1kxacyvLosRHza1i`

**Comment by Huzaifa696** [comment]
For reference, the first two points will be divided into the following three PRs.

1. SVM changes to allow conflicting entries
2. Allow conflicting entries in the account locking function
3. Replay changes to allow self-conflicting entries

And the unit tests for the first two PRs will be as follows:
Cases 1:
1. no conflicts (easy - just basic sanity check)
2. 2 transfers with same fee-payer (valid in both)
3. 2 transfers with same fee-payer (only enough to pay first fee)
4. 2 transfers with same fee-payer (cannot pay 2nd fee because of first transfer)
5. 2 duplicate transactions
6. 2 nonced txs using same nonce

Cases 2:

1. no conflicting locks
2. batch write-write conflict. no outstanding conflicting lock
3. batch write-read conflict. no outstanding conflicting lock
4. batch write-write conflict. outstanding (read/write lock)
5. batch write-read conflict. outstanding (read/write lock)

**Comment by Huzaifa696** [comment]
Just pushed the first two PRs regarding the SVM changes [[link](https://github.com/anza-xyz/agave/pull/1468)] and the account locking changes [[link](https://github.com/anza-xyz/agave/pull/1624)] to allow self-conflicting entries. While these are being reviewed, we try to finalize the third PR.

**Comment by talalim** [comment]
Suggested changes for both PR1,2 incorporated for review.

**Comment by Huzaifa696** [comment]
Initial implementation of [PR3](https://github.com/anza-xyz/agave/pull/2256) (regarding replay side to allow self conflicting entries) code changes is done.
Implementation of its benches and unit tests is WIP.

@apfitzge
